src/e2e-test/java/io/cdap/plugin/utils/JsonUtils.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.utils;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonSyntaxException;
+import org.apache.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * JsonUtils contains the helper functions.
+ */
+public class JsonUtils {
+  private static final Logger logger = Logger.getLogger(JsonUtils.class);
+
+  public static Map<String, String> covertKeyValueJsonArrayToMap(String json) {
+    Map<String, String> keyValueMap = new HashMap<>();
+    try {
+      JsonArray jsonArray = (JsonArray) new JsonParser().parse(json);
+      for (JsonElement jsonElement : jsonArray) {
+        keyValueMap.put(jsonElement.getAsJsonObject().get("key").getAsString(),
+                        jsonElement.getAsJsonObject().get("value").getAsString());
+      }
+    } catch (JsonSyntaxException e) {
+      logger.error("JsonUtils : Error while converting to Map - " + e);
+    } catch (Exception e) {
+      logger.error("JsonUtils : Error while converting to Map - key or value node not present in input json - " + e);
+    }
+    return keyValueMap;
+  }
+
+  public static int countJsonNodeSize(String json, String node) {
+    int size = 0;
+    JsonObject jsonObject = (JsonObject) new JsonParser().parse(json);
+    try {
+      JsonArray dataObject = jsonObject.getAsJsonArray(node);
```

Comment:
would it be possible to first check if it's a json array before calling getAsJsonArray to avoid the exception handling logic.
i.e. something like jsonObject.getAsJsonPrimitive().isJsonArray()